### PR TITLE
Fixed convertMsgToXMessage to handle whatsapp delivery report.

### PR DESCRIPTION
**Pull request by chinmoy12c**
Fixed convertMsgToXMessage to handle whatsapp delivery report.
Added unit tests.

### CODE REVIEW

File: packages/adapters/gupshup-whatsapp/dist/index.d.mts
Comment (ChakshuGautam) [inline packages/adapters/gupshup-whatsapp/dist/index.d.mts:1]: Let's remove `dist` folder and use [prepack lifecycle script](https://docs.npmjs.com/cli/v9/using-npm/scripts#life-cycle-scripts) for publishing.

Comment (ChakshuGautam) [reply packages/adapters/gupshup-whatsapp/dist/index.d.mts:1]: Also let's add release and publish to npm through GitHub Release tags only. GHA triggered on that.

File: packages/adapters/gupshup-whatsapp/src/gupshupWhatsappAdapter.spec.ts [hunk 1]
```diff
@@ -173,4 +173,28 @@ describe('gupshup whatsapp adapter', () => {
   //   expect(actualParametersPassed).toBe(expectedParameters);
   // })
 
+  it ("Convert whatsapp report to XMessage", async () => {
+    const mockReport = {
+      response: "[{\"srcAddr\":\"TESTSM\",\"extra\":\"Samagra\",\"channel\":\"WHATSAPP\",\"externalId\":\"5057936233376494042-daf67a98-e3a3-4f02-8d0b-02bd41ba3aae\",\"cause\":\"SUCCESS\",\"errorCode\":\"000\",\"destAddr\":\"919305097128\",\"eventType\":\"DELIVERED\",\"eventTs\":\"1702464614000\"}]",
+    }
+    const expectedXMessage = {
+      to: { userID: 'admin' },
+      from: { userID: '9305097128' },
```
Comment (ChakshuGautam) [inline packages/adapters/gupshup-whatsapp/src/gupshupWhatsappAdapter.spec.ts:182]: Let's remove PII everywhere.

File: packages/adapters/gupshup-whatsapp/dist/index.d.mts
Comment (chinmoy12c) [reply packages/adapters/gupshup-whatsapp/dist/index.d.mts:1]: Already removed dist.
Will pick GHA on separate PR.

**Comment by chinmoy12c** [comment]
XMessage package needs to be pushed to npm to accommodate the new `REPORT` field in `MessageType`. Once that is done the build should pass. @prtkjakhar

**Review by prtkjakhar — APPROVED** [review]

**Pull request merged**